n3fit/src/n3fit/backends/keras_backend/MetaModel.py

```diff
@@ -53,121 +53,87 @@ def _default_loss(y_true, y_pred): # pylint: disable=unused-argument
         self.has_dataset = False
+        self.required_slots = set()
 
-        input_list = input_tensors
-        output_list = output_tensors
-
-        if isinstance(input_list, dict):
-            # if this is a dictionary, convert it to a list for now
-            input_list = input_tensors.values()
-        elif not isinstance(input_list, list):
-            # if it is not a dict but also not a list, make it into a 1-element list and pray
-            input_list = [input_list]
-
-        if isinstance(output_list, dict):
-            # if this is a dictionary, convert it to a list for now
-            output_list = output_tensors.values()
-        elif not isinstance(output_list, list):
-            # if it is not a dict but also not a list, make it into a 1-element list and pray
-            output_list = [output_list]
-
-        # Note: there used to be two possible options when creating a model:
-        # - Give placeholder tensors (for which the content will be given at run time)
-        # - Give tensors with content* (for which the content is stored with the model
-        # *this option was dropped at some point by TF, the code below keeps this behaviour
-        # We will store within the model the following quantities:
-        #   -> x_in: arrays containing the x-input to the model
-        #   -> tensors_in: when the x-input is known at compile time, we store a reference to the tensor
-        # We pass TensorFlow a dictionary {k: tensor} containing placeholders which will be automatically filled
-        # whenever x_in/tensor_in is known at compile time
+        #input_list = input_tensors
+        #output_list = output_tensors
+
+        if input_values is None:
+            input_values = {}
 
         x_in = {}
-        tensors_in = {}
-        input_dict = {}
-        for input_tensor in input_list:
-            # If the input contains a tensor_content, store it to use at predict/fit/eval times
-            # otherwise, put a placeholder None as it will come from the outside
-            name = input_tensor.name.rsplit(":", 1)[0]
-            input_dict[name] = input_tensor
-            try:
-                x_in[name] = op.numpy_to_tensor(input_tensor.tensor_content)
-                tensors_in[name] = input_tensor
-            except AttributeError:
-                x_in[name] = None
-                tensors_in[name] = None
-
-        super().__init__(input_dict, output_list, **kwargs)
+        # Go over the inputs. If we can deduce a constant value, either because
+        # it is set in input_values or because it has a tensor_content, we
+        # store it. Otherwise we mark the input as required when making
+        # predictions.
+        for k, v in input_tensors.items():
+            if k in input_values:
+                x_in[k] = input_values[k]
+            elif hasattr(v, "tensor_content"):
+                x_in[k] = v.tensor_content
```

Comment:
You should keep the `op.numpy_to_tensor` here.

The reason for that is having one single place where to cast the numpy array to other types (if we want to do `double` instead of `float` or whatever).